shared/Microsoft.Extensions.CopyOnWriteDictionary.Sources/CopyOnWriteDictionary.cs

```diff
@@ -13,6 +13,10 @@ internal class CopyOnWriteDictionary<TKey, TValue> : IDictionary<TKey, TValue>
         private readonly IEqualityComparer<TKey> _comparer;
         private IDictionary<TKey, TValue> _innerDictionary;
 
+        public CopyOnWriteDictionary(IDictionary<TKey, TValue> sourceDictionary)
```

Comment:
Yup - we intentionally did not add this oveload to avoid accidentally forgetting to pass in the comparer when you owned the inner `Dictionary<TKey, TValue>`. Given that it's a .Sources package, we could optionally have it accept a `Dictionary` (concrete type) and do the right thing when no comparer is specified.